desks-web-app/src/components/booking-components/WeekOverviewWindow.vue

```diff
@@ -53,14 +83,29 @@ export default defineComponent({
   methods: {
 
     ...mapActions("defaults", ["fetchBookingDefaults"]),
-    ...mapActions("bookings", ["fetchBookingsByRoomAndWeek"]),
+    ...mapActions("bookings", ["fetchBookingsByRoomAndTimespan"]),
 
     isDisabled: (startOfDay: Moment): boolean => startOfDay.isBefore(moment().startOf("day")),
 
-    calculateNthDayOfWeek(n: number): Moment {
+    calculateNthNextDay(n: number): Moment {
       return moment(this.startOfWeek).add(n, "day")
     },
 
+    isMonday(day: Moment) {
+      return day.isoWeekday() === 1;
+    },
+
+
+    isWeekend(day: Moment) {
+      return day.isoWeekday() > 5;
+    },
+
+    getNext5WorkingDays(): Moment[] {
+      const days = new Array<Moment>(7);
+      for(var i = 0; i < 7; i++) days[i] = this.calculateNthNextDay(i);
+      return days.filter(d => !this.isWeekend(d));
+    },
```

Comment:
Maybe move those date related methods to a plugin to be able  to easily reuse them in other Components.

Reply:
Refactored the whole date / week handling. remove duplicated code and the necessity for similar functions. window is now only the window and gets all dates as and array.